fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/domain/JobExecutionRepository.java

```diff
@@ -165,30 +170,85 @@ public void updateJobStatusToFailed(Long stuckJobId, String partitionerStepName)
                 """, Map.of("status", FAILED.name(), "jobExecutionId", stuckJobId));
     }
 
-    public List<Long> getRunningJobsIdsByExecutionParameter(String jobName, String jobCustomParamKeyName, String parameterKeyName,
+    public LocalDate getBusinessDateOfRunningJobByExecutionParameter(String jobName, String jobCustomParamKeyName, String parameterKeyName,
             String parameterValue) {
-        final StringBuilder sqlStatementBuilder = new StringBuilder();
-        String jsonString = gson.toJson(new JobParameterDTO(parameterKeyName, parameterValue));
-        sqlStatementBuilder.append(
-                "SELECT bje.JOB_EXECUTION_ID FROM BATCH_JOB_INSTANCE bji INNER JOIN BATCH_JOB_EXECUTION bje ON bji.JOB_INSTANCE_ID = bje.JOB_INSTANCE_ID INNER JOIN BATCH_JOB_EXECUTION_PARAMS bjep ON bje.JOB_EXECUTION_ID = bjep.JOB_EXECUTION_ID"
-                        + " WHERE bje.STATUS IN (:statuses) AND bji.JOB_NAME = :jobName AND bjep.PARAMETER_NAME = :jobCustomParamKeyName AND "
-                        + sqlGenerator.castInteger("bjep.PARAMETER_VALUE") + " IN (" + getSubQueryForCustomJobParameters()
-                        + ") AND bje.JOB_INSTANCE_ID NOT IN (SELECT bje.JOB_INSTANCE_ID FROM BATCH_JOB_INSTANCE bji INNER JOIN BATCH_JOB_EXECUTION bje ON bji.JOB_INSTANCE_ID = bje.JOB_INSTANCE_ID"
-                        + " WHERE bje.STATUS = :completedStatus AND bji.JOB_NAME = :jobName)");
-        return namedParameterJdbcTemplate.queryForList(
-                sqlStatementBuilder.toString(), Map.of("statuses", List.of(STARTED.name(), STARTING.name()), "jobName", jobName,
-                        "completedStatus", COMPLETED.name(), "jobCustomParamKeyName", jobCustomParamKeyName, "jsonString", jsonString),
-                Long.class);
-    }
-
-    private String getSubQueryForCustomJobParameters() {
-        if (databaseTypeResolver.isMySQL()) {
-            return "SELECT cjp.id FROM batch_custom_job_parameters cjp WHERE JSON_CONTAINS(cjp.parameter_json,:jsonString)";
-        } else if (databaseTypeResolver.isPostgreSQL()) {
-            return "SELECT cjp.id FROM (SELECT id,json_array_elements(parameter_json) AS json_data FROM batch_custom_job_parameters) AS cjp WHERE (cjp.json_data ::jsonb @> :jsonString ::jsonb)";
-        } else {
-            throw new IllegalStateException("Database type is not supported for json query " + databaseTypeResolver.databaseType());
+        String businessDateInString;
+        try {
+            if (databaseTypeResolver.isPostgreSQL()) {
+                businessDateInString = namedParameterJdbcTemplate.queryForObject("""
+                        SELECT "parameterValue" FROM json_populate_recordset(null::record,
```

Comment:
Can Select J -> ‘parameterValue’ From… be used here?

Reply:
yes it can